code/game/g_client.c

```diff
@@ -1138,36 +1138,82 @@ void ClientUserinfoChanged( int clientNum )
     // Enforce the identities
     oldidentity = client->pers.identity;
 
-    if( level.gametypeData->teams )
-    {
-        s = Info_ValueForKey ( userinfo, "team_identity" );
+    if (!(client->sess.changeCount == 10 && client->sess.idLimit > level.time) ||
+        ((isCurrentGametype(GT_HNS) || isCurrentGametype(GT_HNZ)) && client->sess.team == TEAM_BLUE)) {
 
-        // Lookup the identity by name and if it cant be found then pick a random one
-        client->pers.identity = BG_FindIdentity ( s );
+        if (level.gametypeData->teams && !isCurrentGametype(GT_HNS) && !isCurrentGametype(GT_HNZ)) {
+            const char *teamIdentity = Info_ValueForKey(userinfo, "team_identity");
+            client->pers.identity = BG_FindIdentity(teamIdentity);
 
-        if ( team != TEAM_SPECTATOR )
-        {
-            // No identity or a team mismatch means they dont get to be that skin
-            if ( !client->pers.identity || Q_stricmp ( level.gametypeTeam[team], client->pers.identity->mTeam ) )
-            {
-                // Get first matching team identity
-                client->pers.identity = BG_FindTeamIdentity ( level.gametypeTeam[team], -1 );
+            if (team != TEAM_SPECTATOR) {
+                if (!client->pers.identity || Q_stricmp(level.gametypeTeam[team], client->pers.identity->mTeam)) {
+                    client->pers.identity = BG_FindTeamIdentity(level.gametypeTeam[team], -1);
+                }
+            } else {
+                client->pers.identity = BG_FindTeamIdentity(level.gametypeTeam[TEAM_RED], 0);
             }
-        }
-        else
-        {
-            // Spectators are going to have to choose one of the two team skins and
-            // the chance of them having the proper one in team_identity is slim, so just
-            // give them a model they may use later
-            client->pers.identity = BG_FindTeamIdentity ( level.gametypeTeam[TEAM_RED], 0 );
-        }
-    }
-    else
-    {
-        s = Info_ValueForKey ( userinfo, "identity" );
 
-        // Lookup the identity by name and if it cant be found then pick a random one
-        client->pers.identity = BG_FindIdentity ( s );
+        } else {
+            const char *identityName = Info_ValueForKey(userinfo, "identity");
+
+            if (isCurrentGametype(GT_HNS)) {
+                if (client->sess.team == TEAM_BLUE) {
+                    if (!client->pers.identity ||
+                        !(strstr(client->pers.identity->mName, "NPC_Swiss_Police") ||
+                        strstr(client->pers.identity->mName, "NPC_Honor_Guard") ||
+                        strstr(client->pers.identity->mName, "NPC_Sebastian_Jenzer"))) {
+
+                        int randomIndex = irand(0, 4);
+                        int identityOptions[] = {60, 61, 62, 103, 102};
```

Comment:
Please define them as enum variables, 60, 61, 62, ... they say nothing in codespace.